notebooks/stitches_dev/nearest_neighbor_matching.R

```diff
@@ -43,26 +45,42 @@ internal_dist <- function(fx_pt, dx_pt, archivedata){
   # this returns the first minimum run into, which is not how we are going to want to do it, 
   # we will want some way to keep track of the min and combine to have different realizations 
   # or have some random generation. But for now I think that this is probably sufficent. 
-  index <- which.min(dist$dist_l2) 
-  
-  # if there are mulitple matches then an error should be thrown! Why 
-  # is this not happening for the historiccal period? The ensemble members of 
-  # different experiments should be identical to one another! 
-  if(length(index) > 1){
-    stop("more than one identical match found!")
+  #
+  # probably don't actually need the if-statement to treat tol=0 separately; in theory, the 
+  # else condition would return the nearest neighbor for tol=0. But just in case of rounding
+  # issues, keeping it separate for now to be sure we can replicate previous behavior.
+  if (tol == 0) { 
+    index <- which.min(dist$dist_l2) 
+    
+    # if there are mulitple matches then an error should be thrown! Why 
+    # is this not happening for the historiccal period? The ensemble members of 
+    # different experiments should be identical to one another! 
```

Comment:
Is it not throwing the error because the dist_l2 is not exactly 0, I've run into issues before where 
x - x = 1e-20 or something ridiculously tiny which we humans would think of as being 0 but the computer doesn't so it fails x -x == 0?

Reply:
I'm not sure what you're referring to here. For the `tol == 0`, it's a user specified input and not something calculated, so you'll never get  a 1e-20 kind of case because R is pretty good at 'you explicitly said 0, I'm keeping it 0'. 
If you're referring to the multiple matches error, dunno

Reply:
Oh I guess I was referring to the inline comment about how an error message should be triggered but wasn't

Reply:
I'm not sure - that's holdover from the original matching function so I haven't thought about it at all.

Reply:
👌 then let's continue to not think about it